packages/manager/modules/adp/src/deploy/deploy.controller.js

```diff
@@ -0,0 +1,300 @@
+   * ##################################################################### */
+
+  /**
+   * Checks if General Configurations are complete
+   *
+   * @returns a boolean indicating the state of general configuration step
+   */
+  isGeneralConfigComplete() {
+    return this.vRack && this.vRack.data && this.vRack.data.id
+      && this.selectedSshKey && this.selectedSshKey.id;
+  }
+
+  /**
+   * Saves the required data before the step is finalized
+   */
+  submitGeneralInformation() {
+    console.log('selectedPublicCloud', this.selectedPublicCloud);
+    this.adp.clusterType = this.selectedCapability.version;
+    this.adp.public_cloud = this.selectedPublicCloud.project_id;
+    this.adp.ssh_key = this.selectedSshKey.id;
+  }
+
+  /**
+   * Checks if the master password and the confirm password matches
+   *
+   * @returns a boolean indicating whether the passwords match
+   */
+  checkPasswordMatch() {
+    return this.masterPasswordConfirm === this.adp.master_password;
+  }
+
+  /** #####################################################################
+      REGIONS AND DATACENTER
+   * ##################################################################### */
+
+  /**
+   * fetch all ADP supported regions
+   *
+   * @param {*} publicCloudServiceName public cloud service name
+   * @returns array of vRack objects
+   */
+  fetchRegionDetails(regionCodes) {
+    this.regions = this.cucControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.adpService.getRegionDetails(
+        regionCodes,
+      )
+        .catch(error => this.cucServiceHelper.errorHandler('adp_get_regions_error')(error)),
+    });
+    return this.regions.load();
+  }
+
+  initRegionsAndDatacenters() {
+    // fetch regions
+    this.fetchRegionDetails(get(this.selectedCapability, 'available_region'));
+    this.initRegionsByDatacenter();
+    this.initRegionsByContinent();
+  }
+
+  initRegionsByDatacenter() {
+    this.regionsByDatacenter = this.cucControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.regions
+        .promise
+        .then((regions) => {
+          const regionsByDatacenter = groupBy(regions, 'macroRegion.code');
+          const groupedRegions = map(regionsByDatacenter, (microRegions) => {
+            const region = cloneDeep(microRegions[0]);
+            region.dataCenters = microRegions;
+            delete region.microRegion;
+            delete region.disabled;
+            return region;
+          });
+          return groupedRegions;
+        }),
+    });
+    return this.regionsByDatacenter.load();
+  }
+
+  initRegionsByContinent() {
+    this.regionsByContinent = this.cucControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.regionsByDatacenter
+        .promise
+        .then(regions => groupBy(regions, 'continent')),
+    });
+    return this.regionsByContinent.load();
+  }
+
+  submitRegionAndDatacenter() {
+    console.log('region', this.selectedRegion);
```

Comment:
You should probably remove those `console.log` once the PR will be considered as ready :)